File: dp-ui/src/api/system/dataDictionary.ts
import request from '@/api/request'

export interface DataDictionary {
  id?: number
  dictType: string
  dictCode: string
  dictLabel: string
  dictValue: string
  sortOrder?: number
  isDefault?: boolean
  status?: number
  remark?: string
  createTime?: string
  updateTime?: string
}

/** 分页查询 */
export const getDataDictionaryList = (params?: {
  page?: number; pageSize?: number; dictType?: string; keyword?: string; status?: number
}) => {
  return request.get('/data-dictionary/list', { params })
}

/** 按字典类型查询（下拉用） */
export const getDictByType = (dictType: string) => {
  return request.get(`/data-dictionary/type/${dictType}`)
}

/** 获取所有字典类型 */
export const getDictTypes = () => {
  return request.get('/data-dictionary/types')
}

/** 获取详情 */
export const getDataDictionaryById = (id: number) => {
  return request.get(`/data-dictionary/${id}`)
}

/** 创建 */
export const createDataDictionary = (data: DataDictionary) => {
  return request.post('/data-dictionary', data)
}

/** 更新 */
export const updateDataDictionary = (id: number, data: DataDictionary) => {
  return request.put(`/data-dictionary/${id}`, data)
}

/** 删除 */
export const deleteDataDictionary = (id: number) => {
  return request.delete(`/data-dictionary/${id}`)
}

/** 批量删除 */
export const batchDeleteDataDictionary = (ids: number[]) => {
  return request.delete('/data-dictionary/batch', { data: ids })
}

/** 批量获取字典映射（报表字段值翻译用） */
export const getDictMappings = (types: string) => {
  return request.get('/data-dictionary/mappings', { params: { types } })
}

File: dp-core/src/main/java/com/dataplatform/data/entity/DataDictionary.java
package com.dataplatform.data.entity;

import lombok.Data;
import java.time.LocalDateTime;

/**
 * 数据字典实体
 */
@Data
public class DataDictionary {

    private Long id;

    /** 字典类型 */
    private String dictType;

    /** 字典编码 */
    private String dictCode;

    /** 字典标签 */
    private String dictLabel;

    /** 字典值 */
    private String dictValue;

    /** 排序 */
    private Integer sortOrder;

    /** 是否默认 */
    private Boolean isDefault;

    /** 状态: 0-禁用 1-启用 */
    private Integer status;

    /** 备注 */
    private String remark;

    /** 创建时间 */
    private LocalDateTime createTime;

    /** 更新时间 */
    private LocalDateTime updateTime;
}
